docs/tools/testing/testing_greentea.md

```diff
@@ -178,6 +314,191 @@ This detects your attached target and drives the test. If you need to rerun the
+Test Case:
+    Name: mbed-os-features-cellular-tests-api-cellular_device
+    Path: ./mbed-os/features/cellular/TESTS/api/cellular_device
+
+...
+```
+
+#### Executing all tests
+
+The default action of Greentea using `mbed test` is to execute all tests that are found. You can also add `-v` to make the output more verbose.
+
+#### Limiting tests
+
+You can select test cases by name using the `-n` argument. This command executes all tests named `tests-mbedmicro-rtos-mbed-mail` from all builds in the test specification:
+
+```
+$ mbed test -n tests-mbedmicro-rtos-mbed-mail
+```
+
+When using the `-n` argument, you can use the `*` character at the end of a test name to match all tests that share a prefix. This command executes all tests that start with `tests-mbedmicro-rtos`:
+
+```
+$ mbed test -n tests-mbedmicro-rtos-*
+```
+
+You can use a comma (`,`) to separate test names (argument `-n`) and build names (argument `-t`). This command executes the tests `tests-mbedmicro-rtos-mbed-mail` and `tests-mbed_drivers-c_strings` for the `K64F-ARM` and `K64F-GCC_ARM` builds in the test specification:
+
+```
+$ mbed test -n tests-mbedmicro-rtos-mbed-mail,tests-mbed_drivers-c_strings -t K64F-ARM,K64F-GCC_ARM
+```
+
+#### Selecting platforms
+
+You can limit which boards Greentea uses for testing by using the `--use-tids` argument.
+
+```
+$ mbed test --use-tids 02400203C3423E603EBEC3D8,024002031E031E6AE3FFE3D2
+```
+
+Where `02400203C3423E603EBEC3D8` and `024002031E031E6AE3FFE3D` are the target IDs of platforms attached to your system.
+
+You can view target IDs using Mbed CLI:
+
+```
+$ mbed detect
++--------------+---------------------+------------+------------+-------------------------+
+|platform_name |platform_name_unique |mount_point |serial_port |target_id                |
++--------------+---------------------+------------+------------+-------------------------+
+|K64F          |K64F[0]              |E:          |COM160      |024002031E031E6AE3FFE3D2 |
+|K64F          |K64F[1]              |F:          |COM162      |02400203C3423E603EBEC3D8 |
+|LPC1768       |LPC1768[0]           |G:          |COM5        |1010ac87cfc4f23c4c57438d |
++--------------+---------------------+------------+------------+-------------------------+
+```
+
+In this case, you won't test one target, the LPC1768.
+
+#### Creating reports
+
+Greentea supports a number of report formats.
+
+##### HTML
+
+This creates an interactive HTML page with test results and logs.
+
+```
+mbed test --report-html html_report.html
+```
+
+##### JUnit
+
+This creates an XML JUnit report, which you can use with popular Continuous Integration software, such as [Jenkins](https://jenkins.io/index.html).
+
+```
+mbed test --report-junit junit_report.xml
+```
+
+##### JSON
+
+This creates a general JSON report.
+
+```
+mbed test --report-json json_report.json
+```
+
+##### Plain text
+
+This creates a human-friendly text summary of the test run.
+
+```
+mbed test --report-text text_report.text
```

Comment:
All of these report commands would need the `--run` command added first, otherwise it would throw an exception:

```
$ mbed test -n tests-mbed_functional-callback -v --report-json report.json
[mbed] Working path "C:\Users\bridan01\onedrive_arm\Documents\dev\mbed-os" (program)
[mbed] Exec "c:\python27\python.exe -u C:\Users\bridan01\onedrive_arm\Documents\dev\mbed-os\tools\test.py -t GCC_ARM -m K64F --source . --build .\BUILD\tests\K64F\GCC_ARM --test-spec .\BUILD\tests\K64F\GCC_ARM\test_spec.json --build-data .\BUILD\tests\K64F\GCC_ARM\build_data.json -n tests-mbed_functional-callback -v --greentea --report-json report.json" in "C:\Users\bridan01\onedrive_arm\Documents\dev\mbed-os"
usage: test.py [-h] [-m MCU] [--custom-targets CUSTOM_TARGETS_DIRECTORY]
               [-t TOOLCHAIN] [--color] [--cflags CFLAGS]
               [--asmflags ASMFLAGS] [--ldflags LDFLAGS] [-c]
               [--profile PROFILE] [--app-config APP_CONFIG] [-D MACROS]
               [-j JOBS] [--source SOURCE_DIR] [--build BUILD_DIR] [-l]
               [-p PATHS] [-f FORMAT] [--continue-on-build-fail] [-n NAMES]
               [--test-config TEST_CONFIG] [--test-spec TEST_SPEC]
               [--build-report-junit BUILD_REPORT_JUNIT]
               [--build-data BUILD_DATA] [-v] [--stats-depth STATS_DEPTH]
               [--ignore IGNORE] [--icetea] [--greentea]
test.py: error: unrecognized arguments: --report-json report.json
[mbed] ERROR: "c:\python27\python.exe" returned error.
       Code: 2
       Path: "C:\Users\bridan01\onedrive_arm\Documents\dev\mbed-os"
       Command: "c:\python27\python.exe -u C:\Users\bridan01\onedrive_arm\Documents\dev\mbed-os\tools\test.py -t GCC_ARM -m K64F --source . --build .\BUILD\tests\K64F\GCC_ARM --test-spec .\BUILD\tests\K64F\GCC_ARM\test_spec.json --build-data .\BUILD\tests\K64F\GCC_ARM\build_data.json -n tests-mbed_functional-callback -v --greentea --report-json report.json"
       Tip: You could retry the last command with "-v" flag for verbose output
---
```